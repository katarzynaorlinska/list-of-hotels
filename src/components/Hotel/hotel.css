.hotel{
    max-width: 60vw;
    border: 1px solid #7A7D80;
    border-radius: 10px;
    background-color: white;
    margin: 10vh auto;
}

.hotel__container{
    display: flex;
}

.hotel__name{
    padding: 20px;
    text-align: center;
    font-weight: bold;
    font-size: 20px;
    color: #7cb3ae;
}

.hotel__photo--big{
    max-width: 360px;
    padding-left: 5px;
    padding-bottom: 5px;
}

.hotel__leftBox--small{
    display: flex;
   }

.hotel__photo--small{
    max-width: 175px;
    padding: 5px;
}

.hotel__leftBox{
    display: flex;
    flex-direction: column;
    align-items: baseline;
    padding: 20px;
}

.hotel__leftBox img{
    display: block;
}

.hotel__rightBox{
    display: flex;
    flex-direction: column;
    justify-content: left;
    padding-top: 100px;
}

.hotel__rightBox p{
    display: block;
    margin: 0;
    padding: 5px;
}

.hotel_rightBox--content{
    color: #7cb3ae;
}

@media screen and (max-width: 860px) {
    .hotel__leftBox--small{
        flex-direction: column;
    }

    .hotel__photo{
        max-width: 300px;
    }
  }

  @media screen and (max-width: 730px){
    .hotel__container{
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }
    .hotel__rightBox{
        padding: 20px;
    }
  }

  @media screen and (max-width: 470px){
    .hotel__photo{
        max-width: 150px;
    }
  }