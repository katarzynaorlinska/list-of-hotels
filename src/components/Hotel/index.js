import { useParams } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPhone, faMailBulk } from "@fortawesome/free-solid-svg-icons";

import "./hotel.css";

function Hotel(props) {
  const { id } = useParams();
  const item = props.items.find((element) => id === element.id.toString());

  if(!item){
    return <span>No hotel</span>
  }

   return <div>
      <div className="hotel">
        <div className="hotel__name">{item.name}</div>
        <div className="hotel__container">
          <div className="hotel__leftBox">
            <img src={item.images[0]} alt="hotel" className="hotel__photo hotel__photo--big"></img>
            <div className="hotel__leftBox--small">
              <img src={item.images[1]} alt="hotel" className="hotel__photo hotel__photo--small"></img>
              <img src={item.images[2]} alt="hotel" className="hotel__photo hotel__photo--small"></img>
              </div>
          </div>
          <div className="hotel__rightBox">
            <p><b>Adress:</b> {item.location.address}, {item.location.city}</p>
            <p><b>Check in:</b> {item.checkIn.from} - {item.checkIn.to}</p>
            <p><b>Check out:</b> {item.checkOut.from} - {item.checkOut.to}</p>
            <p><FontAwesomeIcon icon={faPhone} className="hotel_rightBox--content"/> {item.contact.phoneNumber}</p>
            <p><FontAwesomeIcon icon={faMailBulk} className="hotel_rightBox--content" /> {item.contact.email}</p>
          </div>
        </div>
      </div>
    </div>;
}

export default Hotel;
