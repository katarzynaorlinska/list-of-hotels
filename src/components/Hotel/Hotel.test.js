import { create } from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { useParams } from "react-router-dom";
import mockResponse from "../../mock/listHotel.json";
import Hotel from "./index";

jest.mock("react-router-dom", () => ({
  useParams: jest.fn(),
}));
jest.mock("@fortawesome/react-fontawesome", () => ({
  FontAwesomeIcon: () => "FontAwesomeIcon",
}));

test("renders no hotel if no hotel find", () => {
  useParams.mockImplementation(() => ({
    id: "10000000000",
  }));
  render(<Hotel items={mockResponse.hotels} />);
  const noHotel = screen.getByText("No hotel");
  expect(noHotel).toBeInTheDocument();
});

test("renders correctly", () => {
  useParams.mockImplementation(() => ({
    id: "12321",
  }));
  const tree = create(<Hotel items={mockResponse.hotels} />).toJSON();
  expect(tree).toMatchSnapshot();
});
