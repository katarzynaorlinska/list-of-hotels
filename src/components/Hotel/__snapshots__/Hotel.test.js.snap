// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="hotel"
  >
    <div
      className="hotel__name"
    >
      Cfajka
    </div>
    <div
      className="hotel__container"
    >
      <div
        className="hotel__leftBox"
      >
        <img
          alt="hotel"
          className="hotel__photo hotel__photo--big"
          src="https://aff.bstatic.com/images/hotel/max500/102/102202664.jpg"
        />
        <div
          className="hotel__leftBox--small"
        >
          <img
            alt="hotel"
            className="hotel__photo hotel__photo--small"
            src="https://aff.bstatic.com/images/hotel/max500/102/102202628.jpg"
          />
          <img
            alt="hotel"
            className="hotel__photo hotel__photo--small"
            src="https://aff.bstatic.com/images/hotel/max500/102/102202639.jpg"
          />
        </div>
      </div>
      <div
        className="hotel__rightBox"
      >
        <p>
          <b>
            Adress:
          </b>
           
          6 Hercules Road
          , 
          London
        </p>
        <p>
          <b>
            Check in:
          </b>
           
          14:00
           - 
          20:00
        </p>
        <p>
          <b>
            Check out:
          </b>
           
          07:00
           - 
          10:00
        </p>
        <p>
          FontAwesomeIcon
           
          +39 24322342
        </p>
        <p>
          FontAwesomeIcon
           
          park_plaza@gmail.com
        </p>
      </div>
    </div>
  </div>
</div>
`;
